### DISCUSSION TITLE
WEB_VAULT_ENABLED can not be changed from ENV_FILE, but it is implied that it can be

### DISCUSSION THREAD

**Discussion by slonkazoid** [dani-garcia/vaultwarden#4622] [Q&A] [answered]
### Subject of the issue

`WEB_VAULT_ENABLED` can not be changed from `ENV_FILE`, but it is implied that it can be

### Deployment environment

* vaultwarden version: 1.30.5-r2
* Install method: apk add vaultwarden vaultwarden-web-vault
* Clients used:  not applicable
* Reverse proxy and version:  Caddy 2.6.2
* MySQL/MariaDB or PostgreSQL version:  postgresql 16.3
* Other relevant details: not applicable

### Steps to reproduce

1. Install vaultwarden and the web vault
2. Set `WEB_VAULT_ENABLED` to `true` in `ENV_FILE`
3. Go to /admin/diagnostics
4. "Web Vault is disabled"

### Expected behaviour

It should be more clear that you can not enable the web vault from the env file,
or alternatively, it should let you enable the web vault from the env file.

### Actual behaviour

The web vault feature is disabled even if you set the option, and you are not told why.

### Troubleshooting data

not applicable

**Comment by BlackDex** [comment]
The web-vault can be set to enabled via env just fine. 
You should also be sure the web-vault files exist and are accessible for Vaultwarden.

Also, if you ever changed the config via the admin interface before, there might be a `config.json` in the data directory. If that is the case, that overrules any env variable.

**Reply by slonkazoid** [reply]
I checked for all of these. It can be enabled through the environment just fine, but not through the file `ENV_FILE` points to. Everything else works fine in `ENV_FILE`, just not `WEB_VAULT_ENABLED`. To work around this I just exported `WEB_VAULT_ENABLED=true` in my openrc service definition

**Reply by slonkazoid** [reply]
Not to sound rude but I do not get why this was converted into a discussion as this is obviously problematic

**Reply by stefan0xC** [reply, marked as answer]
@slonkazoid given that you used `apk` to install Vaultwarden, you are using a [third party package](https://github.com/dani-garcia/vaultwarden/wiki/Third-party-packages), so it's not an issue we can address.

If you check the [configuration file](https://git.alpinelinux.org/aports/tree/community/vaultwarden/vaultwarden.confd?h=3.20-stable#n34) that is shipped by the alpine package (which you presumably use), you will see that it will `export WEB_VAULT_ENABLED=false`. If this is read into the environment (presumably automatically by OpenRC) the value will not be overriden by a change in the `ENV_FILE` because variables that already exist in the environment will be preserved by [`dotenvy::from_path`](https://docs.rs/dotenvy/0.15.7/dotenvy/fn.from_path.html). And we can / will not change this to [`from_path_override`](https://docs.rs/dotenvy/0.15.7/dotenvy/fn.from_path_override.html) because changing the order of precedence might break existing setups.

**Comment by slonkazoid** [comment]
I haven't read into the configuration system too much but I suspect this might be caused by the check to add / route being evaluated before `ENV_FILE` is parsed

**Reply by stefan0xC** [reply]
Please read https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview because as explained above the issue you ran into has nothing to do with your suspicion and at least as far as I can tell more with downstream defaults (e.g. setting an environment variable which they expect you to override in their config file).

**Comment by slava0008** [comment]
For AlpineLinux edit the file  /etc/conf.d/vaultwarden
Replace export WEB_VAULT_ENABLED=false with export WEB_VAULT_ENABLED=true
